.changeset/spicy-worms-jam.md

```diff
@@ -0,0 +1,6 @@
+---
+'@graphql-codegen/visitor-plugin-common': minor
+'@graphql-codegen/typescript-resolvers': minor
+---
+
+Add resolversNonOptionalTypename config
```

Comment:
Can you please add more information, and and usage example on how it alters the code?

Reply:
Yes! I've updated it here: https://github.com/dotansimha/graphql-code-generator/pull/9146/commits/27af429788178d48442772d4cb88fc96e319a58c